src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/dto/decoderstats/DecoderInfo.java

```diff
@@ -464,16 +557,77 @@ public void setPreviewIntervalSec(String previewIntervalSec) {
 	 *
 	 * @return value of {@link #quadMode}
 	 */
-	public String getQuadMode() {
-		return quadMode;
+	/**
+	 * Retrieves {@code {@link #quadMode}}
+	 *
+	 * @return value of {@link #quadMode}
+	 */
+	public QuadMode getQuadMode() {
+		if (this.quadMode != null) {
+			for (QuadMode quadMode : QuadMode.values()) {
+				if (quadMode.getCode().equals(this.quadMode)) {
+					return quadMode;
+				}
+			}
+		}
+		return QuadMode.NORMAL;
 	}
 
 	/**
 	 * Sets {@code quadMode}
 	 *
-	 * @param quadMode the {@code java.lang.String} field
+	 * @param quadMode the {@code java.lang.Integer} field
 	 */
-	public void setQuadMode(String quadMode) {
+	public void setQuadMode(Integer quadMode) {
 		this.quadMode = quadMode;
 	}
+
+	/**
+	 * This method is used to create request body as Json for decoder controlling start/stop/update
+	 * @return String json request body
+	 */
+	public String jsonRequest() {
+		return '{' +
+				"\"id\":" + id +
+				",\"streamId\":" + streamId +
+				",\"state\":" + state +
+				",\"latency\":" + latency +
+				",\"stillImage\":" + stillImage +
+				",\"stillImageDelay\":" + stillImageDelay +
+				",\"bufferingMode\":" + bufferingMode +
+				",\"bufferingDelay\":" + bufferingDelay +
+				",\"multisyncBufferingDelay\":" + multisyncBufferingDelay +
+				",\"hdrDynamicRange\":" + hdrDynamicRange +
+				",\"output1\":" + output1 +
+				",\"output2\":" + output2 +
+				",\"output3\":" + output3 +
+				",\"output4\":" + output4 +
+				",\"outputFrameRate\":" + outputFrameRate +
+				",\"quadMode\":" + quadMode +
+				'}';
+	}
+
+	public boolean deepEquals(Object o) {
```

Comment:
Please override equals and hashcode. It's completely design for this.

Reply:
I fixed it